### PR TITLE
[IMP] password_security: add flag to enable/disable password security policy

### PR DESCRIPTION
**Pull request by ajaniszewska-dev**
image
image

### CODE REVIEW

File: password_security/models/res_users.py [hunk 1]
```diff
@@ -55,16 +55,17 @@ def write(self, vals):
...
+                    "password_upper": company_id.password_upper,
+                    "password_numeric": company_id.password_numeric,
+                    "password_special": company_id.password_special,
+                    "password_length": company_id.password_length,
+                    "password_estimate": company_id.password_estimate,
+                }
+            )
         return data
 
     def _check_password_policy(self, passwords):
```
Comment (sebalix) [inline password_security/models/res_users.py:72]: This std method should check the new flag as well

**Comment by sebalix** [comment]
NOTE: this flag is required do not get unrelated tests failing when a user is created with a weak password.
In general I think it's better to propose an option to enable a feature if this one could break existing features/flows.

File: password_security/models/res_users.py (hunk 1 above)
Comment (thomaspaulb) [reply password_security/models/res_users.py:72]: @sebalix This comment seems to be open still?

Comment (sebalix) [reply password_security/models/res_users.py:72]: Hello @thomaspaulb it is now fixed

**Comment by sebalix** [comment]
@ajaniszewska-dev @thomaspaulb I rebased the PR and rework a bit its content to reduce the diff to its minimum, ready for another review

**Review by sebalix — APPROVED** [review]

File: password_security/models/res_company.py [hunk 2]
```diff
@@ -9,6 +9,7 @@
 class ResCompany(models.Model):
     _inherit = "res.company"
 
+    password_policy_enabled = fields.Boolean(default=False)
```
Comment (simahawk) [inline password_security/models/res_company.py:12]: I guess whoever installed this module already, expects it to be on by default, right?
If yes, we need a migration script to set the flag to True on upgrade.
N'est pas?

**Review by simahawk — COMMENTED** [review]
Migration script probably needed.

File: password_security/models/res_company.py (hunk 2 above)
Comment (sebalix) [reply password_security/models/res_company.py:12]: You are right! I pushed a migration script (tested locally on a customer project).

**Review by simahawk — APPROVED** [review]

**Pull request merged**